sql/driver/driver.go

```diff
@@ -41,6 +41,7 @@ func (d *roachDriver) Open(dsn string) (driver.Conn, error) {
 	}
 	ctx := &base.Context{}
 	ctx.InitDefaults()
+	ctx.Insecure = u.Scheme == "http"
```

Comment:
This shouldn't be needed, there are two separate registered senders: one for http, the other for https. They each specify a function that sets the Insecure flag from the scheme. See http_sender.go:Init()
This is a verbatim copy of the client/sender functionality. If the context doesn't get properly set, we should figure out why.